test/test_common/utility.h

```diff
@@ -843,6 +843,33 @@ class ConditionalInitializer {
   bool ready_ ABSL_GUARDED_BY(mutex_){false};
 };
 
+namespace Tracing {
+
+class TestTraceContextImpl : public Tracing::TraceContext {
+public:
+  TestTraceContextImpl(const std::initializer_list<std::pair<std::string, std::string>>& values) {
+    for (const auto& value : values) {
+      context_map_[value.first] = value.second;
+    }
+  }
+
+  absl::optional<absl::string_view> getTraceContext(absl::string_view key) const override {
+    auto iter = context_map_.find(std::string(key));
```

Comment:
```suggestion
    auto iter = context_map_.find(key);
```

absl::flat_hash_map support heterogeneous lookup

Reply:
get it.